opal/mca/common/ucx/common_ucx.c

```diff
@@ -126,6 +127,59 @@ OPAL_DECLSPEC void opal_common_ucx_mca_deregister(void)
     opal_output_close(opal_common_ucx.output);
 }
 
+OPAL_DECLSPEC int opal_common_ucx_priority(ucp_context_h context, int min, int max)
+{
+#if HAVE_DECL_OPEN_MEMSTREAM
+    static const char *transports[] = {
+        "cuda_ipc", "ud_verbs", "rc_verbs", "dc_mlx5"
+    };
+    char *match, *ptr, *buffer;
+    char needle[32];
+    FILE *stream;
+    size_t size;
+    size_t i;
+
+    if (min >= max) {
+        /* no need to check anything - minimal priority is already high enough */
+        return min;
+    }
+
+    stream = open_memstream(&buffer, &size);
+    if (stream == NULL) {
+        MCA_COMMON_UCX_VERBOSE(1, "failed to open memory stream for ucx info (%s), "
+                               "setting default ucx priority to %d",
+                               strerror(errno), min);
+        return min;
+    }
+
+    ucp_context_print_info(context, stream);
+
+    fclose(stream);
+
+    for (i = 0; i < sizeof(transports) / sizeof(transports[0]); ++i) {
+        /* Match "resource 6  :  md 5  dev 4  flags -- rc_verbs/mlx5_0:1" */
+        snprintf(needle, sizeof(needle), " %s/", transports[i]);
+        match = strstr(buffer, needle);
```

Comment:
can't we search just transport name without `/`?

Reply:
wanted to have a more strict check so it would not match against partial words

Reply:
ok, maybe add `/mlx` then? To avoid the case when ud is supported for some non-mlnx HCA which may have better support in other pmls/btls